rpm/fingertip.in

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+MACHINES="/home/$USER/.cache/fingertip/machines"
+# Setup
+if [ ! -d "$MACHINES" ]; then
+    echo "The machines filesystem not found. Should I create one?"
+    echo -e "Requires at least 20 GB free space in /home partition (y/N): "
+    read -n 1 -r
+    if [[ ! $REPLY =~ ^[Yy]$ ]]; then
+        exit
+    fi
+    mkdir -p $MACHINES
+    fallocate -l 20G $MACHINES/for-machines.xfs  # or more
+    mkfs.xfs -m reflink=1 $MACHINES/for-machines.xfs
+    sudo mount -o loop $MACHINES/for-machines.xfs $MACHINES
+    sudo chown $USER $MACHINES
```

Comment:
echo "fingertip cleanup machines 12h" >> /home/$USER/.cache/fingertip/cleanup
chmod +x /home/$USER/.cache/fingertip/cleanup
systemd-run --user --on-active 12h /home/$USER/.cache/fingertip/cleanup

what do you think about adding this?

Reply:
Looks reasonable. But see the comments above. This will have to go through some rework and the setup will most probably land inside of the python code based on the snippets Alex posted so this would make sense to land also in there. I hope I will be able to look into that in coming days.

Reply:
duplicating PM discussion here:

* setup would probably move to `main.py`
* I'm not convinced that we need an extra script for it. @sahanaprasad07 made a good point that it will make it more flexible in case we later decide that we want to do more than that one command, but then it might be easier to call a dedicated python function (I propose `cleanup.periodical` and change it when we want to)
* not sure what'd work better, many on-active or a single on-calendar.

Reply:
@sahanaprasad07 Having a second look, why we do not use directly `systemd-run --user --on-active 12h fingertip cleanup machines 12h` directly? Creation of temporary executable file under home directory sounds to me superfluous.

Reply:
@Jakuje: that's also what I suggested initally. But now I agree that if we'll want to extend it in the future to also clean downloads or logs, or change this 12h after-expiration-lingering period to, say, 4h, we'll have to perform quite an awkward sequence of actions to reconfigure the old timer. It's easier to make it trigger something that we can easily reconfigure.

Reply:
Sorry, I did not have loaded the previous comment (I noticed it in the mail, but then I had old tab which was not refreshed). I like the idea of having `fingertip cleanup periodical` called from the timer.